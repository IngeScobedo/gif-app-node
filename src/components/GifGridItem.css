@import url("https://fonts.googleapis.com/css2?family=ZCOOL+KuaiLe&display=swap");
.gif-title {
  font-family: "ZCOOL KuaiLe", cursive;
}

.gif-card {
  background-color: white;
  margin: 10px;
  display: flex;
  flex-direction: column;
  border-radius: 15px;
  justify-content: center;
  align-content: center;
  overflow: hidden;
  border: 1px solid black;
  border-radius: 15px;
  text-align: center;
}
.gif-card-image {
  width: auto;
  max-height: 170px;
}
