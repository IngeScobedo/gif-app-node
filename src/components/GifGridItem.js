import PropTypes from 'prop-types';
import React from 'react'
import './GifGridItem.css'

const GifGridItem = ({ id, title, url }) => {
    return (
        <div className="gif-card animate__animated animate__backInDown" >
            <img 
                src={url} 
                alt={title}
                className="gif-card-image"
                />
            <p className='gif-title'>{title}</p>
        </div>
    )
}

GifGridItem.propTypes = {
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired,
}
export default GifGridItem