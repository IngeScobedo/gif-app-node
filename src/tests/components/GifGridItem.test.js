import { shallow } from "enzyme";
import GifGridItem from "../../components/GifGridItem";

describe("Testing - <GifGridItem/> ", () => {
    let title = 'GifMeme';
    let url = 'https://media2.giphy.com/media/liBsVeLILcyaY/giphy.gif?cid=e2157891xehcho4qeilo5iaijtazd6hkv6ooyrzxaxl2j88h&rid=giphy.gif&ct=g'
  const wrapper = shallow(<GifGridItem 
      title={title}
      url={url}
      />);

  test("Debe mostrar el componente correctamente", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('Debe Tener un parrafo con el titulo', () => {
      const p = wrapper.find('p').text();
      expect(p).toBe(title)
  })
  
  test('Debe tener la imagen desde la url enviada y el alt como el titulo', () => {
      let img = wrapper.find('img');
      expect(img.prop('src')).toBe(url);
      expect(img.prop('alt')).toBe(title);
  })
  
  test('Debe tener 3 clases en total', () => {
      let classExp = 'animate__backInDown';
      let div = wrapper.find('div').prop('className');
      expect(div.includes(classExp)).toBe(true)
    })
  
});
