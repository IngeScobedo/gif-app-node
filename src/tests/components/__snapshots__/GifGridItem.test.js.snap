// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing - <GifGridItem/>  Debe mostrar el componente correctamente 1`] = `
<div
  className="gif-card animate__animated animate__backInDown"
>
  <img
    alt="GifMeme"
    className="gif-card-image"
    src="https://media2.giphy.com/media/liBsVeLILcyaY/giphy.gif?cid=e2157891xehcho4qeilo5iaijtazd6hkv6ooyrzxaxl2j88h&rid=giphy.gif&ct=g"
  />
  <p
    className="gif-title"
  >
    GifMeme
  </p>
</div>
`;
